File: api/src/main/java/org/jukeboxmc/plugin/perms/api/JPerms.java
package org.jukeboxmc.plugin.perms.api;

import org.jukeboxmc.plugin.perms.api.manager.GroupManager;
import org.jukeboxmc.plugin.perms.api.manager.PermissionManager;

/**
 * @author Kaooot
 * @version 1.0
 */
public interface JPerms {

    /**
     * Retrieves the permission manager which is responsible for managing the permissions of a
     * certain user
     *
     * @return a fresh {@link org.jukeboxmc.plugin.perms.api.manager.PermissionManager}
     */
    PermissionManager permissionManager();

    /**
     * Retrieves the group manager which is responsible for managing available groups and the groups
     * of a certain user
     *
     * @return a fresh {@link org.jukeboxmc.plugin.perms.api.manager.GroupManager}
     */
    GroupManager groupManager();
}

File: api/src/main/kotlin/org/jukeboxmc/plugin/perms/api/JPerms.kt
package org.jukeboxmc.plugin.perms.api

import org.jukeboxmc.plugin.perms.api.manager.GroupManager
import org.jukeboxmc.plugin.perms.api.manager.PermissionManager

/**
 * @author Kaooot
 * @version 1.0
 */
interface JPerms {

    /**
     * Retrieves the permission manager which is responsible for managing the permissions of a
     * certain user
     *
     * @return a fresh [org.jukeboxmc.plugin.perms.api.manager.PermissionManager]
     */
    fun permissionManager(): PermissionManager

    /**
     * Retrieves the group manager which is responsible for managing available groups and the groups
     * of a certain user
     *
     * @return a fresh [org.jukeboxmc.plugin.perms.api.manager.GroupManager]
     */
    fun groupManager(): GroupManager
}